### PR TITLE
fix device for get_ppl

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

fix device for get_ppl

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

None

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by lvhan028 — APPROVED** [review]

**Comment by irexyc** [comment]
# concat logits. Shape is [bsz, seq_len, vocab_size]
logits = torch.cat(logits, dim=1)

这里用cuda存挺危险的。src里面的logits也得改成cpu的
